content/en/docs/tasks/administer-cluster/memory-manager.md

```diff
@@ -0,0 +1,394 @@
+
+```shell
+--feature-gates=MemoryManager=true 
+--kube-reserved=cpu=4,memory=4Gi 
+--system-reserved=cpu=1,memory=1Gi 
+--memory-manager-policy=static 
+--reserved-memory 0:memory=3Gi --reserved-memory 1:memory=2148Mi
+```
+Let us validate the configuration above:
+1. `kube-reserved + system-reserved + eviction-hard(default) = reserved-memory(0) + reserved-memory(1)`
+2. `4GiB + 1GiB + 100MiB = 3GiB + 2148MiB`
+3. `5120MiB + 100MiB = 3072MiB + 2148MiB`
+4. `5220MiB = 5220MiB` (which is correct)
+
+## Placing a Pod in the Guaranteed QoS class
+
+If the selected policy is anything other than `none`, the Memory Manager identifies pods that are in the `Guaranteed` QoS class. The Memory Manager provides specific topology hints to the Topology Manager for each `Guaranteed` pod. For pods in a QoS class other than `Guaranteed`, the Memory Manager provides default topology hints to the Topology Manager.
+
+The following excerpts from pod manifests assign a pod to the `Guaranteed` QoS class.
+
+Pod with integer CPU(s) runs in the `Guaranteed` QoS class, when `requests` are equal to `limits`:
+
+```yaml
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    resources:
+      limits:
+        memory: "200Mi"
+        cpu: "2"
+        example.com/device: "1"
+      requests:
+        memory: "200Mi"
+        cpu: "2"
+        example.com/device: "1"
+```
+
+Also, a pod sharing CPU(s) runs in the `Guaranteed` QoS class, when `requests` are equal to `limits`.
+
+```yaml
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    resources:
+      limits:
+        memory: "200Mi"
+        cpu: "300m"
+        example.com/device: "1"
+      requests:
+        memory: "200Mi"
+        cpu: "300m"
+        example.com/device: "1"
+```
+
+Notice that both CPU and memory requests must be specified for a Pod to lend it to Guaranteed QoS class.
+
+## Troubleshooting
+
+The following means can be used to troubleshoot the reason why a pod could not be deployed or became rejected at a node:
+- pod status - indicates topology affinity errors
+- system logs - include valuable information for debugging, e.g., about generated hints
+- state file - the dump of internal state of the Memory Manager (includes [Node Map and Memory Maps][2]) 
+
+### Pod status (TopologyAffinityError) {#TopologyAffinityError}
+
+This error typically occurs in the following situations:
+* a node has not enough resources available to satisfy the pod's request
+* the pod's request is rejected due to particular Topology Manager policy constraints 
+
+The error appears in the status of a pod:
+```shell
+# kubectl get pods
+NAME         READY   STATUS                  RESTARTS   AGE
+guaranteed   0/1     TopologyAffinityError   0          113s
+```
+
+Use `kubectl describe pod <id>` or `kubectl get events` to obtain detailed error message:
+```shell
+Warning  TopologyAffinityError  10m   kubelet, dell8  Resources cannot be allocated with Topology locality
+```
+
+### System logs

```

Comment:
the logs change as part of structured logging conversion, i am not sure we need to document at this level?

Reply:
@derekwaynecarr Thanks for this tip. According to https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/ :
> We have also updated many logging calls such that over 99% of logging volume in a typical deployment are now migrated to the structured format.

I removed samples and only described what can be found in the logs.
If structured logging format samples are required, we can provide them in the future.

CC @cynepco3hahue Still, some changes in code might be required to fully support structured logging.